src/utils/transaction.ts

```diff
@@ -934,6 +1137,210 @@ export const buildCrossChainAction = async (
+            const erc20Contract = sdk.registerContract<ERC20TokenContract>(
+              'erc20Contract',
+              abi,
+              routeToUSDC.fromToken.address
+            );
+            const approvalTransactionRequest = erc20Contract?.encodeApprove?.(
+              transactions[0].to,
+              routeToUSDC.fromAmount
+            );
+            if (!approvalTransactionRequest || !approvalTransactionRequest.to) {
+              return { errorMessage: 'Failed build bridge approval transaction!' };
+            }
+
+            const approvalTransaction = {
+              to: approvalTransactionRequest.to,
+              data: approvalTransactionRequest.data,
+              value: 0,
+              createTimestamp,
+              status: CROSS_CHAIN_ACTION_STATUS.UNSENT,
+            };
+
+            transactions = [approvalTransaction, ...transactions];
+          }
+
+          console.log('sksksk', offer1, offer2);
+
+          // Swap 1 Start //
+          destinationTxns = [
+            ...destinationTxns,
+            ...offer1.transactions.map((transaction) => ({
+              ...transaction,
+              chainId: CHAIN_ID.XDAI,
+              createTimestamp,
+              status: CROSS_CHAIN_ACTION_STATUS.UNSENT,
+            })),
+          ];
+
+          // SWAP 1 ENDS
+
+          // Swap 2 Start
+
+          destinationTxns = [
+            ...destinationTxns,
+            ...offer2.transactions.map((transaction) => ({
+              ...transaction,
+              chainId: CHAIN_ID.XDAI,
+              createTimestamp,
+              status: CROSS_CHAIN_ACTION_STATUS.UNSENT,
+            })),
+          ];
+
+          // SWAP 2 ENDS
+
+          const addressToSendTo = receiverAddress ?? sdk.state.accountAddress;
+
+          const honeySwapTransaction = await honeyswapLP(
+            sdk,
+            offer1.receiveAmount,
+            toToken1.address,
+            offer2.receiveAmount,
+            toToken2.address,
+            offer1,
+            offer2,
+            addressToSendTo
+          );
+
+          if (honeySwapTransaction.errorMessage) return { errorMessage: honeySwapTransaction.errorMessage };
+
+          console.log('destinationTxns1Swap+', destinationTxns);
+
+          if (honeySwapTransaction.result?.transactions?.length) {
+            destinationTxns = [...destinationTxns, ...honeySwapTransaction.result?.transactions];
+          }
+
+          const preview = {
+            fromChainId,
+            toChainId: CHAIN_ID.XDAI,
+            fromAsset: {
+              address: fromAssetAddress,
+              decimals: fromAssetDecimals,
+              symbol: fromAssetSymbol,
+              amount: amount,
+              iconUrl: fromAssetIconUrl,
+            },
+            amount: ethers.utils.parseUnits(amount ?? '0', 6),
+            toAsset: {
+              address: '0xDDAfbb505ad214D7b80b1f830fcCc89B60fb7A83',
```

Comment:
This is the USDC address again right? I thought you declared a variable further up - regardless I think having the export would resolve this, means you don't have to comment on every instance you use this as well